src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -3177,6 +3219,64 @@ ITermDispatch::StringHandler AdaptDispatch::_CreateDrcsPassthroughHandler(const
     return nullptr;
 }
 
+// Method Description:
+// - DECDMAC - Defines a string of characters as a macro that can later be
+//   invoked with a DECINVM sequence.
+// Arguments:
+// - macroId - a number to identify the macro when invoked.
+// - deleteControl - what gets deleted before loading the new macro data.
+// - encoding - whether the data is encoded as plain text or hex digits.
+// Return Value:
+// - a function to receive the macro data or nullptr if parameters are invalid.
+ITermDispatch::StringHandler AdaptDispatch::DefineMacro(const VTInt macroId,
+                                                        const DispatchTypes::MacroDeleteControl deleteControl,
+                                                        const DispatchTypes::MacroEncoding encoding)
+{
+    if (!_macroBuffer)
+    {
+        _macroBuffer = std::make_shared<MacroBuffer>();
+    }
+
+    if (_macroBuffer->InitParser(macroId, deleteControl, encoding))
+    {
+        return [&](const auto ch) {
+            return _macroBuffer->ParseDefinition(ch);
+        };
+    }
+
+    return nullptr;
+}
+
+// Method Description:
+// - DECINVM - Invokes a previously defined macro, executing the macro content
+//   as if it had been received directly from the host.
+// Arguments:
+// - macroId - the id number of the macro to be invoked.
+// Return Value:
+// - True
+bool AdaptDispatch::InvokeMacro(const VTInt macroId)
+{
+    if (_macroBuffer)
+    {
+        const auto macroSequence = _macroBuffer->GetMacroSequence(macroId);
+        if (!macroSequence.empty())
+        {
```

Comment:
`GetMacroSequence` and `InvokeMacroSequence` are only used in a single place (here). I think they can be merged by making `InvokeMacroSequence` accept the `macroId` instead of a string-view. That way this if condition won't be necessary either anymore.

Reply:
Hmmm... I remember splitting them up for a reason, but I can't recall why. I think perhaps I had intended to use the `GetMacroSequence` in the unit tests at one point. I'll try merging them again and see what happens.

Reply:
I've merged those now. That's much better. Thanks for bringing that up.